### PR TITLE
Renumber Dimensions enum

### PR DESCRIPTION
**Pull request by audunska**
I am happy with this compromise.

Let's do it before we start using it.

### CODE REVIEW

**Comment by audunska** [comment]
Protobufs _requires_ the constants to start from 0. I added some junk to make it compile, but it might not be ideal..

File: cognite/seismic/protos/types.proto [hunk 1]
```diff
@@ -279,6 +279,8 @@ enum IngestionSource {
 }
 
 enum Dimensions {
-    TWO_DEE = 0;
-    THREE_DEE = 1;
+    INVALID_ZERO = 0; // Not used
+    ONE_DEE = 1; // Not implemented
```
Comment (BenChand) [inline cognite/seismic/protos/types.proto:283]: Is it possible to use the `reserved` keyword here for 0 to 1?

Comment (audunska) [reply cognite/seismic/protos/types.proto:283]: I tried, it didn't seem to work

Comment (BenChand) [reply cognite/seismic/protos/types.proto:283]: I guess maybe not if the 0 is the default :/

**Comment by mr-celo** [comment]
I was thinking about this as well when creating the enum. I'm ok with this if you guys are.
My rationale for the PR was that the actual number of the enum was something internal and, as long as the public interface referenced sensible number for the dimensions, we shouldn't really care which underlying number encodes an enum field

**Comment by skairunner** [comment]
my 2 nok: I don't think we need the enum int values to match up with the number of dimensions. The string names corresponding to the internal value is what's meaningful after all, and the internal value is simply to distinguish.

**Comment by audunska** [comment]
I am almost ready to drop this.

I'll just say that we _don't_ in fact use the constant names in the rust code. The enum constant names don't show up, so we have to [refer to the numbers themselves](https://github.com/cognitedata/seismic-resource/blob/97d9612340dd4d61fe6fe823b0749329165abf7a/seismic-service/tests/mutable/register_file_tests.rs#L35). That's what might lead to confusion.

I guess that doesn't happen in the sdk, though.

File: cognite/seismic/protos/types.proto (hunk 1 above)
Comment (amorken) [reply cognite/seismic/protos/types.proto:283]: proto says no. All enums need a zero-valued enum constant. "INVALID" would be a perfectly good choice for that , though.

**Comment by skairunner** [comment]
I don't know if it's worthwhile, but maybe we could write a Rust enum that mirrors the protobuf? The downside is, of course, that it isn't connected to the protobuf's enums and could go out of sync without noticing...

**Comment by amorken** [comment]
> I'll just say that we don't in fact use the constant names in the rust code. The enum constant names don't show up, so we have to [refer to the numbers themselves](https://github.com/cognitedata/seismic-resource/blob/97d9612340dd4d61fe6fe823b0749329165abf7a/seismic-service/tests/mutable/register_file_tests.rs#L35). That's what might lead to confusion.

The fact that we _can_ refer to the enums by ordinal value doesn't mean we _have to_, there are accessor methods. Alright, in initalizer statements like the above we have to, but we could also use the enum name as i32, ie. 'Dimensions::TWO_DEE as i32'.

**Comment by audunska** [comment]
> > I'll just say that we don't in fact use the constant names in the rust code. The enum constant names don't show up, so we have to [refer to the numbers themselves](https://github.com/cognitedata/seismic-resource/blob/97d9612340dd4d61fe6fe823b0749329165abf7a/seismic-service/tests/mutable/register_file_tests.rs#L35). That's what might lead to confusion.
> 
> The fact that we _can_ refer to the enums by ordinal value doesn't mean we _have to_, there are accessor methods. Alright, in initalizer statements like the above we have to, but we could also use the enum name as i32, ie. `Dimensions::TWO_DEE as i32`.

We can? Then that's the solution. I'll close.

**Comment by audunska** [comment]
>  `Dimensions::TWO_DEE as i32`.

Trying to use `proto::Dimensions::THREE_DEE`, I get:
```
error[E0432]: unresolved import `proto::Dimensions::THREE_DEE`
  --> seismic-service/tests/mutable/register_file_tests.rs:10:29
   |
10 |     Crs, DeleteFileRequest, Dimensions::THREE_DEE, EditSurveyRequest, File, Identifier,
   |                             ^^^^^^^^^^^^^^^^^^^^^ no `THREE_DEE` in `Dimensions`

```

**Comment by amorken** [comment]
> > `Dimensions::TWO_DEE as i32`.
> 
> Trying to use `proto::Dimensions::THREE_DEE`, I get:
> 
> ```
> error[E0432]: unresolved import `proto::Dimensions::THREE_DEE`
>   --> seismic-service/tests/mutable/register_file_tests.rs:10:29
>    |
> 10 |     Crs, DeleteFileRequest, Dimensions::THREE_DEE, EditSurveyRequest, File, Identifier,
>    |                             ^^^^^^^^^^^^^^^^^^^^^ no `THREE_DEE` in `Dimensions`
> ```

Prost helpfully case-switches things. Tried Three_Dee or ThreeDee?

**Comment by audunska** [comment]
> Prost helpfully case-switches things. Tried Three_Dee or ThreeDee?

"Helpfully". That did it, thanks

**Comment by amorken** [comment]
> "Helpfully".

Precisely 😁

**Comment by amorken** [comment]
Hmmm. Reopen and merge...? Oh wait, it's deleted. Hmm.

**Pull request closed without merging**